.github/workflows/rust.yml

```diff
@@ -79,20 +78,19 @@ jobs:
         uses: ./.github/actions/install-rust-toolchain
         with:
           toolchain: ${{ matrix.toolchain }}
-
-      - name: Install tools
-        uses: taiki-e/install-action@v2
-        with:
-          tool: cargo-hack@0.5.26,cargo-script@0.2.8,clippy-sarif@0.3.7,sarif-fmt@0.3.7
-
-      - name: Install cargo-make
-        run: cargo install --version 0.36.3 cargo-make
+          working-directory: ${{ matrix.directory }}
+          components: rustfmt,clippy
```

Comment:
Should we maybe extract those from the toml via the python script instead? This way, we only have a single source of truth.

Reply:
partially addressed in #2297